Update enum quoting and default schema use

austindrenski opened the pull request:
## Changes
- If no schema is specified in `.ForNpgsqlHasEnum(...)`, then use the default schema annotation.
- Properly quote schema-qualified enums for literal generation and mapping (override `StoreType`).
- Cache mappings from CLR values to PostgreSQL labels.

Fixes: #554 
Fixes: #593

YohDeadfall commented:
Does this request fixes #623? If not, please fix it in a different PR, not here. This would make it easier to track changes.

austindrenski commented:
Updated to `Npgsql 4.1.0-ci.1270` to use the default name translator from https://github.com/npgsql/npgsql/pull/2137.

austindrenski commented:
Rebased and updated to reflect #621.

austindrenski commented:
> can you please make clearer exactly what this PR covers?

Sure, I'll clean that up today. This PR grew and shifted a bit since I opened it, so the current description is a bit of lazy editing on my part.

> This doesn't seem like the correct EF Core behavior. 

Early on for EF Core, I thought I read some docs that discussed order being relevant in configuration, but I can't seem to find anything like that now. (Could have been a third-party blog, TBH.)

What you describe makes more sense (and matches the internals too). I can go back through and move the default schema detection later in the process.

> BTW does this depend npgsql/npgsql#2121

No, not anymore. An earlier draft was, but I reworked this one to use the simple "split on the first dot" approach while I work through npgsql/npgsql#2121 (which has turned out to be a bit more confusing than I first imagined).

austindrenski commented:
@roji The [description](https://github.com/npgsql/Npgsql.EntityFrameworkCore.PostgreSQL/pull/605#issue-209060244) is up to date, and I've rebased for the current `dev`. I've also rebased #626 which follows this PR with similar fixes for user-defined ranges.

I'd like to see this make it into `v2.2.0`, any chance you can give it a review before the next preview release?

@YohDeadfall Could you also give this another review?

austindrenski commented:
@YohDeadfall Appreciate the review—updated.

austindrenski commented:
I'd like to merge this and #626 tomorrow at 1:00 PM (UTC).

If anyone wants more time for another round of review, just let me know.

/cc @roji @YohDeadfall
